spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/TensorflowToBigDL.scala

```diff
@@ -0,0 +1,1091 @@
+    }
+
+    val buffer = ByteBuffer.wrap(tfTensor.getTensorContent.toByteArray)
+    buffer.order(endian)
+
+    if (classTag[T] == classTag[Float]) {
+      if (tfTensor.getDtype == DataType.DT_FLOAT) {
+        val params = buffer.asFloatBuffer
+        val tmp = new Array[Float](params.capacity())
+        var j = 0
+        while (j < params.capacity()) {
+          tmp(j) = params.get(j)
+          j += 1
+        }
+        Tensor(Storage(tmp), 1, shape).asInstanceOf[Tensor[T]]
+      } else if (tfTensor.getDtype == DataType.DT_INT32) {
+        val params = buffer.asIntBuffer
+        val tmp = new Array[Float](params.capacity())
+        var j = 0
+        while (j < params.capacity()) {
+          tmp(j) = params.get(j)
+          j += 1
+        }
+        Tensor(Storage(tmp), 1, shape).asInstanceOf[Tensor[T]]
+      } else {
+        throw new IllegalArgumentException("Can not convert double to float")
+      }
+    } else if (classTag[T] == classTag[Double]) {
+      if (tfTensor.getDtype == DataType.DT_FLOAT) {
+        val params = buffer.asFloatBuffer
+        val tmp = new Array[Double](params.capacity())
+        var j = 0
+        while (j < params.capacity()) {
+          tmp(j) = params.get(j)
+          j += 1
+        }
+        Tensor(Storage(tmp), 1, shape).asInstanceOf[Tensor[T]]
+      } else if (tfTensor.getDtype == DataType.DT_INT32) {
+        val params = buffer.asIntBuffer
+        val tmp = new Array[Double](params.capacity())
+        var j = 0
+        while (j < params.capacity()) {
+          tmp(j) = params.get(j)
+          j += 1
+        }
+        Tensor(Storage(tmp), 1, shape).asInstanceOf[Tensor[T]]
+      } else if (tfTensor.getDtype == DataType.DT_DOUBLE) {
+        val params = buffer.asDoubleBuffer()
+        val tmp = new Array[Double](params.capacity())
+        var j = 0
+        while (j < params.capacity()) {
+          tmp(j) = params.get(j)
+          j += 1
+        }
+        Tensor(Storage(tmp), 1, shape).asInstanceOf[Tensor[T]]
+      } else {
+        throw new IllegalArgumentException("Data type ${tfTensor.getDtype} is not supported now")
+      }
+    } else {
+      throw new IllegalArgumentException("Only support Float/Double")
+    }
+  }
+
+  private var patternList : ArrayBuffer[TensorflowToBigDL] = {
+    val res = new ArrayBuffer[TensorflowToBigDL]()
+    // ElementWiseMulTF must be after MulTF
+    res.append(
+      FullConnectionTF, DropoutTF, AvgPoolingTF, MaxPoolingTF, ReshapeTF, InputTF,
+      TanhTF, ReluTF, SigmoidTF, Conv2D, Placeholder, SqueezeTF, IdentityTF, ConcatTF,
+      BatchNormTF, AddConstTF1, AddConstTF2, AddTF, SoftMaxTF, MulTF, ElementWiseMulTF,
+      SplitTF, PaddingTF, MeanTF, UnpackTF, StrideSliceTF, ShapeTF, FillTF, PackTF, ConstTF
+    )
+    res
+  }
+
+  sortPattern()
+
+  /**
+   * Sort the pattern list to make sure the graph match first should not be a sub-graph of the graph
+   * match later
+   */
+  private def sortPattern() : Unit = {
```

Comment:
There is a potential issue about this. `MulTF` and `ElementWiseMulTF` pattern have the exact same number of nodes and edges, but `MulTF` should comes before `ElementWiseMulTF`.